expo-boomboomapp/src/components/SongPicker.tsx

```diff
@@ -41,22 +41,25 @@ export function SongPicker({
   debounceTime = DEFAULT_DEBOUNCE_TIME,
 }: SongPickerProps) {
   const spotifyApiService = getGlobalInstance<SpotifyApiServiceI>(
-    ServiceInterface.SpotifyApiServiceI,
+    ServiceInterface.SpotifyApiServiceI
   );
   const languageService = getGlobalInstance<LanguageService>(
-    ServiceInterface.LanguageServiceI,
+    ServiceInterface.LanguageServiceI
   );
   const I18n = languageService.useTranslation();
   const onSearch = (searchString?: string): void => {
-    spotifyApiService
-      .fetchTracksNyName(searchString)
-      .then((tracks): void => {
-        setFetchedSongs(tracks);
-      })
-      .catch((error) => {
-        // TODO handle error better
-        console.log("spotifyApiService : ", error);
-      });
+    if (searchString != undefined && searchString != "") {
+      spotifyApiService
+        .fetchTracksNyName(searchString)
+        .then((tracks): void => {
+          setFetchedSongs([]);
```

Comment:
be carefull

Reply:
About ?

Reply:
you have write setFetchedSongs([]); and setFetchedSongs(tracks.data); next to each other